index.html

```diff
@@ -1915,47 +1915,87 @@ <h3>Defining 'string'</h3>
+
+<p>Because there are multiple ways of working with strings, different terminology has evolved to support the needs of different specifications. Be sure to understand your specification's needs and use the most appropriate and precise terminology. On the Web, there are three types of strings:
+
+<ul>
+	<li>{{USVString}}. Strings based on Unicode <a>code points</a>, also known as </a><a>Unicode Scalar Values</a></li>
+	<li>{{DOMString}}. Strings based on <a>UTF-16</a> <a>code units</a></li>
+	<li>{{ByteString}}. Strings based on bytes in some <a>character encoding form</a> (usually <a>UTF-8</a>)</li>
+</ul></p>
+
+<p>One difference between these different string types is how <a>surrogate</a> <a>code points</a> are handled. In a <code translate="no">USVString</code>, isolated <a>surrogate</a> code points are invalid and implementations are required to replace any found in a string with the Unicode replacment character (<span class="codepoint" translate="no"><bdi lang="und">&#xFFFD;</bdi><code class="uname">U+FFFD REPLACEMENT CHARACTER</code></span>).</p>
+
+<p>By contrast, a {{DOMString}} allows unpaired <a>surrogate</a> <a>code units</a> to appear in a string. An unpaired surrogate can result in processing errors or it must be replaced with the Unicode replacment character (<span class="codepoint" translate="no"><bdi lang="und">&#xFFFD;</bdi><code class="uname">U+FFFD REPLACEMENT CHARACTER</code></span>) when processing the characters inside the string.</p>
+
+<p>In well-formed <a>UTF-8</a>, <a>surrogate</a> <a>code points</a> should not be encoded and isolated surrogates are replaced by the replacement character. <a>Legacy character encodings</a> never encode <a>surrogates</a>.</p>
+
+<p>The [[DOM]], JavaScript (including its APIs and document formats), and certain document formats and platforms use the <a>UTF-16</a> <a>character encoding</a> to store and process string data. The string type based on <a>UTF-16</a> <a>code units</a> is called a {{DOMString}} [[WEBIDL]]. The [[INFRA]] standard defines the term 'string' to mean a {{DOMString}}:</p>
+
+<p class="localdef">A string is a sequence of unsigned 16-bit integers, also known as <a>code units</a>.</p>
+
+<p class="note">[[INFRA]]'s use of the term <a>code unit</a> refers specifically to the <a>UTF-16</a> character encoding's code units, rather than the more general definition of a <a>code unit</a> that can refer to different size values, such as bytes, in any <a>character encoding form</a>.</p>
+
 <div class="req" id="char_string_domstring">
-	<p class="advisement">When designing a web platform feature which operates on strings, use <a href="https://webidl.spec.whatwg.org/#idl-DOMString">DOMString</a> unless you have a specific reason not to.</p>
+	<p class="advisement">When designing a Web platform feature which operates on strings, use <a href="https://webidl.spec.whatwg.org/#idl-DOMString">DOMString</a> unless you have a specific reason not to.</p>
```

Comment:
I suppose we can replace `<a>` with `{{DOMString}}`?

Reply:
Done.